modules/encrypt/classes/Kohana/Encrypt/Engine/Sodium.php

```diff
@@ -64,87 +81,53 @@ public function __construct($config)
      */
     public function encrypt(string $message, string $iv)
     {
-        $value = sodium_crypto_aead_aes256gcm_encrypt($message, '', $iv, $this->_key);
+		$value = call_user_func(
+			'sodium_crypto_aead_' . $this->_cipher . '_encrypt', $message, '', $iv, $this->_key
+		);
 
         if ($value === FALSE)
         {
             // Encryption failed
             return NULL;
         }
 
-        //Base64 encode binary data, otherwise they cannot be
-        //transformed into JSON.
+        //Base64 encode binary data, otherwise they cannot be transformed into JSON.
         $value = base64_encode($value);
         $iv = base64_encode($iv);
 
         $json = json_encode(compact('iv', 'value'));
 
-        if (!is_string($json))
-        {
-            // Encryption failed
-            return NULL;
-        }
-
-        return base64_encode($json);
+		return !is_string($json) ? NULL : base64_encode($json);
     }
 
     /**
      * Decrypts the ciphertext
-     * @param string $ciphertext Ciphertext to be decrypted
+     * @param  string $ciphertext Ciphertext to be decrypted
      * @return null|string
      */
     public function decrypt(string $ciphertext)
     {
         // Convert the data back to binary
         $decode = base64_decode($ciphertext);
-        //Check if base64 decoding succeeded
-        if ($decode === FALSE)
-        {
-            return NULL;
-        }
 
-        $data = json_decode($decode, TRUE);
-
-        //check if json_decode succeeded
-        if (is_null($data))
-        {
-            return NULL;
-        }
-
-        // If the payload is not valid JSON or does not have the proper keys set we will
-        // assume it is invalid and bail out of the routine since we will not be able
-        // to decrypt the given value. We'll also check the MAC for this encryption.
-        if (!$this->valid_payload($data))
-        {
-            // Decryption failed
-            return NULL;
-        }
-
-        $iv = base64_decode($data['iv']);
-        if ($iv === FALSE)
-        {
-            // Invalid base64 data
+		// If the payload is not valid JSON or does not have the proper keys set we will
+		// assume it is invalid and bail out of the routine since we will not be able
+		// to decrypt the given value. We'll also check the MAC for this encryption.
+        if (
+			$decode === FALSE || ($data = json_decode($decode, TRUE)) === NULL ||
```

Comment:
Can we not assign and compare in the same line again. please?

Reply:
Actually this if statement is very hard to debug, can we go back to previous if statements?

Reply:
Yes